### PR TITLE
ECL-631: Accessibility fix

### PR DESCRIPTION
**Pull request by paddynski-opencast**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala [hunk 1]
```diff
@@ -24,14 +24,21 @@ import scala.util.control.Exception.nonFatalCatch
...
 
       override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], String] =
         data.get(key) match {
-          case None                      => Left(Seq(FormError(key, errorKey, args)))
-          case Some(s) if s.trim.isEmpty => Left(Seq(FormError(key, errorKey, args)))
-          case Some(s)                   => Right(s)
+          case None    => Left(Seq(FormError(key, errorKey, args)))
+          case Some(s) =>
+            sanitise(s) match {
+              case "" => Left(Seq(FormError(key, errorKey, args)))
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala:39]: you can use `case _.isEmpty`, using "" is discouraged in FP

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala [hunk 2]
```diff
@@ -24,14 +24,21 @@ import scala.util.control.Exception.nonFatalCatch
...
+              case ss => Right(ss)
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala:40]: what does ss mean?

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Mappings.scala [hunk 3]
```diff
@@ -25,38 +25,44 @@ import java.time.LocalDate
 
 trait Mappings extends Formatters with Constraints {
 
-  protected def text(errorKey: String = "error.required", args: Seq[String] = Seq.empty): FieldMapping[String] =
-    of(stringFormatter(errorKey, args))
+  protected def sanitise(value: String) =
+    value.replaceAll("\\s", "")
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Mappings.scala:29]: `str.filterNot(_.isWhitespace)`

this is a functional way to do it

File: test-common/uk/gov/hmrc/economiccrimelevyregistration/generators/Generators.scala [hunk 4]
```diff
@@ -90,13 +91,13 @@ trait Generators {
   def stringsWithMaxLength(maxLength: Int): Gen[String] =
     for {
       length <- choose(1, maxLength)
-      chars  <- listOfN(length, arbitrary[Char])
+      chars  <- listOfN(length, arbitrary[Char].retryUntil(!_.isWhitespace))
```
Comment (liandra-giacomine) [inline test-common/uk/gov/hmrc/economiccrimelevyregistration/generators/Generators.scala:94]: can probably use     
`chars <- listOfN(length, alphaNumChar)` with .mkString or `str <- alphaNumStr` to get a string

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala [hunk 5]
```diff
@@ -24,14 +24,21 @@ import scala.util.control.Exception.nonFatalCatch
 
 trait Formatters {
 
-  private[mappings] def stringFormatter(errorKey: String, args: Seq[String] = Seq.empty): Formatter[String] =
+  private[mappings] def stringFormatter(
+    errorKey: String,
+    args: Seq[String] = Seq.empty,
+    sanitise: String => String
+  ): Formatter[String] =
     new Formatter[String] {
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala:33]: Do we need a sanitise function here? Seems like we always use .trim for the stringFormatter?

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala (hunk 1 above)
Comment (paddynski-opencast) [reply app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala:39]: Ok Used `case ss if ss.isEmpty` as case `_.IsEmpty` didn't work!

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala (hunk 2 above)
Comment (paddynski-opencast) [reply app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala:40]: Sanitised String

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Mappings.scala (hunk 3 above)
Comment (paddynski-opencast) [reply app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Mappings.scala:29]: Changed the code.

File: test-common/uk/gov/hmrc/economiccrimelevyregistration/generators/Generators.scala (hunk 4 above)
Comment (paddynski-opencast) [reply test-common/uk/gov/hmrc/economiccrimelevyregistration/generators/Generators.scala:94]: Changed the code.

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala (hunk 5 above)
Comment (paddynski-opencast) [reply app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala:33]: We do as in most cases we get rid of all whitespace. In some situations we only trim.

**Pull request merged**